packages/vhd-lib/merge.js

```diff
@@ -49,12 +42,16 @@ module.exports = limitConcurrency(2)(async function merge(
       checkSecondFooter: mergeState === undefined,
     })
     const childVhd = yield openVhd(childHandler, childPath)
+
     if (mergeState === undefined) {
       assert.strictEqual(childVhd.header.blockSize, parentVhd.header.blockSize)
 
       const parentDiskType = parentVhd.footer.diskType
       assert(parentDiskType === DISK_TYPES.DIFFERENCING || parentDiskType === DISK_TYPES.DYNAMIC)
       assert.strictEqual(childVhd.footer.diskType, DISK_TYPES.DIFFERENCING)
+    } else {
+      assert.strictEqual(parentVhd.header.checksum, mergeState.parent.header)
+      assert.strictEqual(childVhd.header.checksum, mergeState.child.header)
```

Comment:
You changed the behavior: what happened if the asserts threw and what happens now?

Reply:
I think  the right comportment is to throw an error, it was the comportment before the major overhaul for s3
https://github.com/vatesfr/xen-orchestra/commit/da0cd0b99cdccc525da39d8b549ee725446a582b#diff-086489f0cd0ea1e1d37f7ab853b50c6eacd10a7fc96ff1100f5c66473958a3a1

Reply:
Fine by me, your correction is good enough :+1: